com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs

```diff
@@ -0,0 +1,252 @@
+using System;
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Jobs;
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.Formats.Alembic.Importer;
+using UnityEngine.Rendering;
+using Unity.Burst;
+
+namespace Scripts.Importer
+{
+    [RequireComponent(typeof(AlembicCurves))]
+    [RequireComponent(typeof(MeshRenderer))]
+    [RequireComponent(typeof(MeshFilter))]
+    [ExecuteInEditMode]
+    public class AlembicCurvesRenderer : MonoBehaviour
+    {
+        AlembicCurves curves;
+        Mesh mesh;
+        [SerializeField] RenderMethod renderMethod;
+        [NonSerialized] RenderMethod prevRenderMethod;
+        ProfilerMarker setMeshProperties = new ProfilerMarker("SetMeshProperties");
+
+
+        enum RenderMethod
+        {
+            Line,
+            Strip
+        }
+
+        void OnEnable()
+        {
+            curves = GetComponent<AlembicCurves>();
+            //curves.OnUpdate += UpdateMesh;
+
+            mesh = new Mesh {hideFlags = HideFlags.DontSave};
+            GetComponent<MeshFilter>().sharedMesh = mesh;
+            var meshRenderer = GetComponent<MeshRenderer>();
+            if (meshRenderer.sharedMaterial == null)
+            {
+                meshRenderer.sharedMaterial = GetDefaultMaterial();
+            }
+            UpdateMesh(curves);
+        }
+
+        void UpdateMesh(AlembicCurves curves)
+        {
+            if (prevRenderMethod != renderMethod)
+            {
+                mesh.Clear();
+            }
+
+            prevRenderMethod = renderMethod;
+
+            switch (renderMethod)
+            {
+                case RenderMethod.Line:
+                    GenerateLineMesh(mesh, curves.Positions, curves.CurvePointCount);
+                    break;
+                case RenderMethod.Strip:
+                    //GeneratePlaneMesh(mesh, new[] {new Vector3(0, 0, 0), new Vector3(0, 1, 0), new Vector3(0, 2, 0)}, new[] {3}, new[] {0.1f, 0.75f, 0.1f});
+                    GeneratePlaneMesh(mesh, curves.Positions, curves.CurvePointCount, curves.Widths);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        void LateUpdate()
+        {
+            UpdateMesh(curves);
+        }
+
+        void OnDisable()
+        {
+            //curves.OnUpdate -= UpdateMesh;
+            DestroyImmediate(mesh);
+        }
+
+        void GeneratePlaneMesh(Mesh theMesh, IReadOnlyList<Vector3> positions, IReadOnlyList<int> curveCounts, IReadOnlyList<float> widths)
```

Comment:
No comments on the algo for the future maintainer?

Reply:
Will remove the algorithm: it's slow and does not look too good.